specification/applicationinsights/resource-manager/microsoft.insights/stable/2015-05-01/componentWorkItemConfigs_API.json

```diff
@@ -0,0 +1,316 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ApplicationInsightsManagementClient",
+    "description": "Azure Application Insights client for work item configurations for a component.",
+    "version": "2015-05-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/components/{resourceName}/WorkItemConfigs": {
+      "get": {
+        "description": "Gets the list work item configurations that exist for the application",
+        "operationId": "WorkItemConfigurations_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/WorkItemConfiguration"
+              }
+            }
+          },
+          "404": {
+            "description": "No configurations exist",
+            "schema": {
+              "$ref": "#/definitions/WorkItemConfigurationError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkItemConfigurationsList": {
+            "$ref": "./examples/WorkItemConfigsGet.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      },
+      "post": {
```

Comment:
put for create. Why is this a POST.

Reply:
Its a POST because the Url path doesn't uniquely define the unique resource. The url path is /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/components/{resourceName}/WorkItemConfigs and  there can be multiple for single resource.

Reply:
POST should not be used for creating resources. Its not idempotent. It breaks consistency around how resources get provisioned in across Azure. Moreover, although its possible to do it via POST, we strongly recommend teams against creating multiple resources in one API call.

Reply:
This api does not create multiple resources in one API  call, it creates a single one only. I agree that it is not idempotent, but did not know this as a requirement. Will keep this guideline for future api versions. Where are api guidelines documented for ARM?